lib/tasks/populate_time_with_timezone.rake

```diff
@@ -4,12 +4,25 @@ namespace :measurements do
     sleep_time = 0.5
     total_processed = 0
     last_maintenance_at = 0
-    maintenance_interval = 200_000_000
-
-    total_to_update = Measurement.where(time_with_time_zone: nil).count
+    maintenance_interval = 50_000_000
+
+    stream_ids_to_update_sql = <<-SQL
+      SELECT DISTINCT stream_id FROM measurements WHERE time_with_time_zone IS NULL
+    SQL
+    stream_ids_to_update = ActiveRecord::Base.connection.execute(stream_ids_to_update_sql).map { |row| row['stream_id'] }
+
+    session_ids_to_update_sql = <<-SQL
+      SELECT DISTINCT session_id FROM streams WHERE id = ANY(ARRAY#{stream_ids_to_update})
+    SQL
+    session_ids_to_update = ActiveRecord::Base.connection.execute(session_ids_to_update_sql).map { |row| row['session_id'] }
+
+    total_to_update_sql = <<-SQL
+      SELECT COUNT(*) FROM measurements WHERE time_with_time_zone IS NULL
+    SQL
+    total_to_update = ActiveRecord::Base.connection.execute(total_to_update_sql).first['count'].to_i
     puts "Total measurements to update: #{total_to_update}"
 
-    Session.find_each(batch_size: 100) do |session|
+    Session.where(id: session_ids_to_update).find_each(batch_size: 100) do |session|
```

Comment:
How about doing it like that?
```
Session.select('streams.id, sessions.time_zone').distinct
.joins(streams: :measurements).where(measurements: { time_with_time_zone:  nil })
```
This way we fetch only data needed for the update - `stream_id` and `time_zone`


I've experienced some weird behaviour with _select_ method, not sure why it's working like that, but from what I tested it seems that the results are correct. Here is how a single record returned by this query looks like:
`#<Session:0x00005562ed5fb428 id: 2494073, time_zone: "Europe/Paris", tag_list: nil>`
- `id` refers to `streams.id`
- I have no idea why `tag_list` is added there 🤷

Reply:
And then having `stream_id`, we could simplify updating measurements:
line 36: `WHERE id IN (SELECT id FROM measurements WHERE stream_id = '#{stream_id})`

From what I've checked there are 2 377 866 measurements per stream at most, so maybe we can do it all at once? WDYT?